pandas/core/internals/managers.py

```diff
@@ -1952,8 +1952,25 @@ def _compare_or_regex_search(a, b, regex=False):
         # GH#29553 avoid deprecation warnings from numpy
         result = False
     else:
+        # GH#32621 use mask to avoid comparing to NAs
+        if is_a_array and not is_b_array:
+            mask = np.reshape(~(isna(a)), a.shape)
+        elif is_b_array and not is_a_array:
+            mask = np.reshape(~(isna(b)), b.shape)
+        elif is_a_array and is_b_array:
+            mask = ~(isna(a) | isna(b))
+
+        if is_a_array:
```

Comment:
can you not do this in the above logic?

Reply:
Hi @jreback, I moved that code block above (https://github.com/pandas-dev/pandas/pull/32890/commits/c32a2cca47fa7e1f3269755f10f376fbfe45bcc1). Can you please confirm that this is what you were expecting? Thanks.